data_validation/src/spark_rapids_validation_tool/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,289 @@
+        return str(dict(sorted_data))
+
+    result = None
+    agg_functions = [min, max, avg, stddev, countDistinct]
+    # if not specified any included_columns, then get all numeric cols and string and map cols
+    excluded_columns_list = [e.strip() for e in e.split(",")]
+    metrics_cols = [i.strip() for i in i.split(",") if i not in excluded_columns_list]
+    if i in ['None', 'all']:
+        metrics_cols = [c.name for c in table_DF.schema.fields if
+                        any(fnmatch.fnmatch(c.dataType.simpleString(), pattern) for pattern in ['*int*', '*decimal*', '*float*', '*double*', 'string', '*map*'])]
+    map_metrics_cols = [c.name for c in table_DF.schema.fields if
+                    any(fnmatch.fnmatch(c.dataType.simpleString(), pattern) for pattern in ['*map*'])]
+    normal_metrics_cols = list(set(metrics_cols) - set(map_metrics_cols))
+    for col in normal_metrics_cols:
+        dfc = spark.createDataFrame(([col],), ["ColumnName"])
+        table1_agg = table_DF.select(
+            [f(col).alias(f.__name__ + t1) for f in
+             agg_functions])
+        tmp_df = dfc.join(table1_agg)
+        if result is None:
+            result = tmp_df
+        else:
+            result = result.union(tmp_df)
+
+    for col in map_metrics_cols:
+        dfc = spark.createDataFrame(([col],), ["ColumnName"])
+        table1_agg = table_DF.select(
+            [f(map_to_string(col)).alias(f.__name__ + t1) for f in
+             agg_functions])
+        tmp_df = dfc.join(table1_agg)
+        if result is None:
+            result = tmp_df
+        else:
+            result = result.union(tmp_df)
+    return result
+
+def metrics_metadata(spark, format, t1, t2, t1p, t2p, pk, i, e, f, p):
+    """
+    The different metadata of each column in each table(min/max/avg/stddev/count_distinct):
+    (If the values are identical, then a specific cell is empty, aka NULL. So we only show differences),
+    the result dataframe should be:
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    |ColumnName| min_A| min_B| max_A| max_B|            avg_A|            avg_B|           stddev_A|           stddev_B|countdist_A|countdist_B|
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
+    |      col1|      |      |12.000|11.000|6.090909090909091|              6.0|  3.477198454346414|    3.3166247903554|           |           |
+    |      col3|      |      |      |      |5.090909090909091|5.181818181818182| 3.1766191290283903|  3.060005941764879|           |           |
+    |      col8|12.340|12.330|      |      |       12.4345455|       12.4336364|0.30031195901474267|0.30064173786328213|          3|          4|
+    +----------+------+------+------+------+-----------------+-----------------+-------------------+-------------------+-----------+-----------+
```

Comment:
nit: showing a sample demo, it will be nice to make it more readable by rounding the float digits to 2 decimal points. Wide lines will fail the pylint test anyway.